server/controllers/userController.js

```diff
@@ -1,112 +1,157 @@
-  const { email, savedPlans } = req.body;
-  if (!email) {
-    return res.status(400).json({ message: "Email is required" });
-  }
-
-  const duplicate = await User.findOne({ email });
-
-  if (duplicate) {
-    return res.status(201).json({ message: "Email already existed" });
-  }
-
-  const user = await User.create({ email, savedPlans });
-
-  if (user) {
-    return res.status(201).json({ message: `User ${email} created` });
-  } else {
-    return res.status(400).json({ message: "Invalid user data received" });
+  try {
+    const { email, savedPlans } = req.body;
+    if (!email) {
+      return res.status(400).json({ message: "Email is required" });
+    }
+
+    const duplicate = await User.findOne({ email });
+
+    if (duplicate) {
+      return res.status(201).json({ message: "Email already existed" });
+    }
+
+    const user = await User.create({ email, savedPlans });
+
+    if (user) {
+      return res.status(201).json({ message: `User ${email} created` });
+    } else {
+      return res.status(400).json({ message: "Invalid user data received" });
+    }
+  } catch (error) {
+    console.error(error);
+    res.status(500).json({ message: "Server error" });
   }
 };
 
 const addPlanToUser = async (req, res) => {
-  const { email, newPlan } = req.body;
-
-  const user = await User.findOne({ email });
-
-  if (!user) {
-    return res.status(404).json({ message: "User not found" });
-  }
-
-  const planExists = user.savedPlans.some((plan) => plan.name === newPlan.name);
-
-  if (planExists) {
-    return res
-      .status(409)
-      .json({ message: "Plan with the same name already exists" });
+  try {
+    const { email } = req.params;
+    const { newPlan } = req.body;
+
+    if (!email || !newPlan) {
+      return res.status(400).json({ message: "Email and plan are required" });
+    }
+
+    const user = await User.findOne({ email });
+
+    if (!user) {
+      return res.status(404).json({ message: "User not found" });
+    }
+
+    const planExists = user.savedPlans.some(
+      (plan) => plan.name === newPlan.name,
+    );
+
+    if (planExists) {
+      return res
+        .status(409)
+        .json({ message: "Plan with the same name already exists" });
+    }
+
+    const updatedUser = await User.findOneAndUpdate(
+      { email },
+      { $push: { savedPlans: newPlan } },
+      { new: true },
+    );
+
+    res
+      .status(201)
+      .json({ message: "Plan added to saved plans", user: updatedUser });
+  } catch (error) {
+    console.error(error);
+    res.status(500).json({ message: "Server error" });
   }
-
-  const updatedUser = await User.findOneAndUpdate(
-    { email },
-    { $push: { savedPlans: newPlan } },
-    { new: true }
-  );
-
-  res
-    .status(201)
-    .json({ message: "Plan added to saved plans", user: updatedUser });
 };
 
 const updatePlanUser = async (req, res) => {
-  const { email, newPlan } = req.body;
+  try {
+    const { email } = req.params;
+    const { newPlan } = req.body;
```

Comment:
I think you get the ``planName`` from ``req.params['planName']`` and not ``req.body``